### ISSUE TITLE
Test failure when upgrading from 4.3.1 to 4.4.0

### ISSUE THREAD

**Issue by matthewwillian** [miguelgrinberg/python-socketio#385]
Coming from usage of Flask-SocketIO. 

```
from api.sockets import socket_io

def test_socketio(app, app_test_client):
    """Test that SocketIO connects."""
    socketio_test_client = socket_io.test_client(
        app, flask_test_client=app_test_client
    )
    assert not socketio_test_client.is_connected()
```

the namespace is defined as follows:

```
from flask_socketio import Namespace, SocketIO

from api.context import get_user

socket_io = SocketIO()  # SocketIO server - initialized within app init


class RootNamespace(Namespace):

    def on_connect(self):
        """Event handler for connecting to a namespace."""
        return get_user() is not None


socket_io.on_namespace(RootNamespace("/"))
```

this unit test fails against 4.4.0 but passes against 4.3.1. did the behavior of `on_connect` get indirectly changed by this version bump?

**Comment by miguelgrinberg** [comment]
Yeah, I'll have to look into this one, it is a side effect of the addition of the `connect_error` event.

**Comment by miguelgrinberg** [comment]
@matthewwillian Could you give the master branch of Flask-SocketIO a try and report back if the issue is corrected?

**Comment by viggyfresh** [comment]
hey @miguelgrinberg just following up on this - the test still fails with release 4.6.0!

**Comment by miguelgrinberg** [comment]
@viggyfresh there is a unit test in this repository that checks `is_connected()` and the test is passing. So I suspect whatever the issue is on your side it is different.

**Comment by viggyfresh** [comment]
@miguelgrinberg sorry for the delay here - I think I tracked down the issue!

We've bumped `python-socketio` up to 4.5.1, but I noticed in the commit that closes this issue (miguelgrinberg/Flask-SocketIO@a05ff51) that the fix was actually in the Flask-SocketIO package, which is still on version 4.2.1 (released August 2019, while the fix commit landed in November).

Any chance you'd be able to cut a release of Flask-SocketIO? Thanks a bunch - and no rush!
